### PR TITLE
feat(deps/stdin): add --from-stdin flag and PipeChecker interface

### PR DESCRIPTION
**Pull request by ShettyGaurav**
**Jira:** [RHOAIENG-66864](https://redhat.atlassian.net/browse/RHOAIENG-66864)

## Description

- Adds `--from-stdin` flag to `deps install`, `components enable/disable`, `lint`, and `migrate run` — reads JSON/YAML config from stdin with CLI flags taking precedence
- Centralises terminal detection in `stdin.CheckPiped` (hard error) and a new `PipeChecker` interface for non-`*os.File` readers in tests
- Adds `stdin.FlagChanged` helper replacing per-command `flagChanged()` methods
- Enforces nil-vs-empty `TargetDeps` semantics: `nil` = bulk mode, `[]string{}` = explicit install-nothing, non-empty = named targets
- Zeroes `TargetDep` after normalization into `TargetDeps` to prevent stale reads post-`Complete()`
- Guards against specifying both a positional dep argument and stdin `deps` field simultaneously

## Testing

- [x] Unit tests pass (`make test`)
- [x] Linter passes (`make lint`)
- [x] New functionality includes tests

## Review checklist

- [x] Code follows project conventions (see docs/coding/)
- [X] Changes are documented if needed

## Summary by CodeRabbit

* **New Features**
  * Enhanced stdin-based configuration via `--from-stdin` (JSON/YAML) across multiple commands, including correct handling of explicit empty dependency lists (“install nothing”).

* **Improvements**
  * Stronger validation to require piped stdin.
  * Consistent CLI-flag precedence over stdin values, with corrected `dry-run`/confirmation behavior.
  * Install behavior and messaging improved for targeted vs empty/no-op cases.

* **Tests**
  * Added and expanded unit tests for stdin parsing, precedence, and install selection edge cases.

### CODE REVIEW

**Comment by ShettyGaurav** [comment]
@coderabbitai review

File: pkg/deps/install.go [hunk 1]
```diff
@@ -158,6 +177,56 @@ func (c *InstallCommand) Complete() error {
...
+	for i, name := range input.Deps {
+		trimmed := strings.TrimSpace(name)
+		if trimmed == "" {
+			return fmt.Errorf("empty dep name at index %d", i)
+		}
+
+		input.Deps[i] = trimmed
+	}
+
+	if input.Deps != nil && (c.TargetDep != "" || len(c.TargetDeps) > 0) {
```
Comment (dayakar349) [inline pkg/deps/install.go:205]: applyStdinInput checks c.TargetDep != "" || len(c.TargetDeps) > 0 before Complete() normalizes TargetDep into TargetDeps. This creates a fragile ordering dependency, if someone reorders the calls in Complete(), the conflict check silently breaks. Consider adding a brief comment documenting that stdin parsing must run before normalization.

Comment (ShettyGaurav) [reply pkg/deps/install.go:205]: Added a comment .

**Review by dayakar349 — APPROVED** [review]
